pkg/query-service/app/querier/querier.go

```diff
@@ -469,7 +470,115 @@ func (q *querier) runClickHouseQueries(ctx context.Context, params *v3.QueryRang
 	return results, errQueriesByName, err
 }
 
+type logsListTsRange struct {
+	Start int64
+	End   int64
+}
+
+const HOUR_NANO = int64(3600000000000)
+
+func getLogsListTsRanges(start, end int64) []logsListTsRange {
+	startNano := utils.GetEpochNanoSecs(start)
+	endNano := utils.GetEpochNanoSecs(end)
+	result := []logsListTsRange{}
+
+	if endNano-startNano > HOUR_NANO {
+		bucket := HOUR_NANO
+		tStartNano := endNano - bucket
+
+		complete := false
+		for {
+			result = append(result, logsListTsRange{Start: tStartNano, End: endNano})
+			if complete {
+				break
+			}
+
+			bucket = bucket * 2
+			endNano = tStartNano
+			tStartNano = tStartNano - bucket
+
+			// break condition
+			if tStartNano <= startNano {
+				complete = true
+				tStartNano = startNano
+			}
+		}
+	}
+	return result
+}
+
+func (q *querier) runLogsListQuery(ctx context.Context, params *v3.QueryRangeParamsV3, keys map[string]v3.AttributeKey, tsRanges []logsListTsRange) ([]*v3.Result, map[string]error, error) {
+	res := make([]*v3.Result, 0)
+	qName := ""
+	pageSize := uint64(0)
+
+	// se we are considering only one query
+	for name, v := range params.CompositeQuery.BuilderQueries {
+		qName = name
+		pageSize = v.PageSize
+	}
```

Comment:
Let's be clear about which query. Should this be the last query? Do we consider enabled/disabled?

Reply:
Makes sense, I have restricted it to run only for queries where there is a single list query.  https://github.com/SigNoz/signoz/pull/5912/commits/803d480596ddb916d88146fc09c9b8997a1d0d8b .


We can come back if we want to support his feature for multiple list queries.